web/src/app/admin/admin-api-keys/AdminAPIKeyForm.tsx

```diff
@@ -0,0 +1,174 @@
+  setAllowFieldsError: (param: boolean) => void
+  create: boolean
+}
+
+export default function AdminAPIKeyForm(
+  props: AdminAPIKeyFormProps,
+): JSX.Element {
+  const { ...containerProps } = props
+  const [expiresAt, setExpiresAt] = useState<string>(
+    DateTime.now().plus({ days: 7 }).toLocaleString({
+      weekday: 'short',
+      month: 'short',
+      day: '2-digit',
+      year: 'numeric',
+      hour: '2-digit',
+      minute: '2-digit',
+      second: '2-digit',
+    }),
+  )
+  const [allowedFields, setAllowedFields] = useState<string[]>([])
+  // handle AllowedFields field option changes: sets selected value to state
+  const handleAutocompleteChange = (
+    event: SyntheticEvent<Element, Event>,
+    value: string[],
+  ): void => {
+    setAllowedFields(value)
+  }
+  // sets GQLAPIKey updated value to state
+  useEffect(() => {
+    const valTemp = props.value
+    valTemp.expiresAt = new Date(expiresAt).toISOString()
+    valTemp.allowedFields = allowedFields
+    props.setAllowFieldsError(valTemp.allowedFields.length <= 0)
+    props.onChange(valTemp)
+  })
+  const { data, loading, error } = useQuery(listGQLFieldsQuery)
+
+  if (error) {
+    return <GenericError error={error.message} />
+  }
+
+  if (loading && !data) {
+    return <Spinner />
+  }
+
+  return (
+    <FormContainer {...containerProps}>
+      <Grid container spacing={2}>
+        <Grid item xs={12}>
+          <FormField
+            fullWidth
+            label='Name'
+            name='name'
+            required
+            component={TextField}
+            value={props.value.name}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <FormField
+            fullWidth
+            label='Description'
+            name='description'
+            multiline
+            rows={4}
+            required
+            component={TextField}
+            charCount={MaxDetailsLength}
+            value={props.value.description}
+            hint='Markdown Supported'
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <FormField
+            fullWidth
+            component={TextField}
+            select
+            required
+            label='Role'
+            name='role'
+            disabled={!props.create}
+            value={props.value.role}
+          >
+            <MenuItem value='user' key='user'>
+              User
+            </MenuItem>
+            <MenuItem value='admin' key='admin'>
+              Admin
+            </MenuItem>
+          </FormField>
+        </Grid>
+        <Grid item xs={12}>
+          {props.create ? (
+            <AdminAPIKeyExpirationField
+              setValue={setExpiresAt}
+              value={expiresAt}
+              create
+            />
+          ) : (
+            <FormField
+              fullWidth
+              component={ISODateTimePicker}
+              name='expiresAt'
+              required
+              disabled
+            />
+          )}
```

Comment:
Can we make `AdminAPIKeyExpirationField` support the `disabled` prop so we don't need a ternary here?

Then this can just become `<AdminAPIKeyExpirationField ... disabled={!create} />`

Reply:
Noted and this has been addressed